plutus-core/cost-model/create-cost-model/BuiltinMemoryModels.hs

```diff
@@ -173,8 +173,8 @@ builtinMemoryModels = BuiltinCostModelBase
   -- paramCaseList
   -- paramCaseData
   , paramDropList                        = Id $ ModelTwoArgumentsConstantCost 4
-  , paramLengthOfArray                   = Id $ ModelOneArgumentConstantCost    99
-  , paramListToArray                     = Id $ ModelOneArgumentConstantCost    99
-  , paramIndexArray                      = Id $ ModelTwoArgumentsConstantCost   99
+  , paramLengthOfArray                   = Id $ ModelOneArgumentConstantCost 10
+  , paramListToArray                     = Id $ ModelOneArgumentLinearInX identityFunction
+  , paramIndexArray                      = Id $ ModelTwoArgumentsConstantCost 32
```

Comment:
@kwxm `10` and `32` here are `magic` numbers I chose because they were chose in similar builtin functions. This is not based on any measurements I did whatsoever.

Reply:
I'm not sure what we should put for the memory costs.  I guess for `ListToArray` we'll need something with a non-zero constant term to account for some header data and then one pointer's worth of memory for each entry.  @effectfully Do you remember how much room a pointer takes up?

For `LengthOfArray` we'll need enough memory to hold one integer, so maybe a constant cost of 1 would do, and for `IndexArray` it'll again be one pointer's worth.

But then again, the current numbers are pretty similar to those for other things like `LengthOfByteString`, so maybe these are OK.  We charge 100 memory units for each CEK step anyway, so that dominates the memory cost and and extra 10 or 20 bytes for a builtin won't make a lot of difference.

Reply:
> @effectfully Do you remember how much room a pointer takes up?

One machine word IIRC.

Reply:
I found [this page](https://wiki.haskell.org/GHC/Memory_Footprint), which says that the memory footprint of a vector is (7+N) words (plus the size of the contents, which we don't care about since they're already on the heap.  Based on that I suggestthe following:

 * Change the memory cost of `listToArray` to `Id $ ModelOneArgumentLinearInX $ OneVariableLinearFunction 7 1`
  * In the three json files, change the  memory model for `listToArray` to match that (this'll save you from having to regenerate the entire cost model):
```
     "memory": {
            "arguments": {
                "intercept": 7,
                "slope": 1
            },
            "type": "linear_in_x"
        }
  ```
   * Update the conformance results.